### PR TITLE
Fix DynamicIndex implementation

### PR DESCRIPTION
**Pull request by StefH**

### CODE REVIEW

File: src/System.Linq.Dynamic.Core/System.Linq.Dynamic.Core.csproj [hunk 1]
```diff
@@ -77,6 +77,7 @@
     <ItemGroup Condition=" '$(TargetFramework)' == 'netstandard1.3' ">
         <PackageReference Include="System.ComponentModel.TypeConverter" Version="4.3.0" />
         <PackageReference Include="System.Dynamic.Runtime" Version="4.3.0" />
+        <PackageReference Include="Microsoft.CSharp" Version="4.3.0" />
```
Comment (hazzik) [inline src/System.Linq.Dynamic.Core/System.Linq.Dynamic.Core.csproj:80]: Do you actually need this reference?

Comment (hazzik) [reply src/System.Linq.Dynamic.Core/System.Linq.Dynamic.Core.csproj:80]: No, it is not needed after f907489ce1c8818f5466a87bdc5377c8938ba5a7 where you exclude `DynamicGetMemberBinder` from netstandard1.3

**Comment by StefH** [comment]
Hello @JonathanMagnan,

Can you or a developer please take a look at this PR ? 

This PR solves issue #448 but in order to fix that I had to remove the **DynamicIndex** logic which was introduced in 1.1.8 and use the original implementation (with a small change) to make sure that #397 keeps working correctly.

For that, I did enable the unittest again:
https://github.com/zzzprojects/System.Linq.Dynamic.Core/pull/480/files#diff-8c64d2f034f0d47405c4723ad04680f885cd71c73a5b0f498d04e95949789348L95

Please review the changes.

File: src/System.Linq.Dynamic.Core/System.Linq.Dynamic.Core.csproj (hunk 1 above)
Comment (StefH) [reply src/System.Linq.Dynamic.Core/System.Linq.Dynamic.Core.csproj]: Thanks. It's removed.

**Comment by hazzik** [comment]
@StefH with following implementation I was able to get a robust solution for my problem as well as un-comment `ExpressionTests_Select_DynamicObjects` test

```
internal class DynamicGetMemberBinder : GetMemberBinder
{
    private static readonly Type DynamicWrapperType = typeof(DynamicWrapper);
    private static readonly ConstructorInfo DynamicWrapperConstructor = DynamicWrapperType.GetConstructor(new[] {typeof(object)});
    private static readonly PropertyInfo DynamicWrapperTypeIndexer = DynamicWrapperType.GetProperty("Item");

    public DynamicGetMemberBinder(string name, [CanBeNull] ParsingConfig config) : base(name, !(config?.IsCaseSensitive == true))
    {
    }

    public override DynamicMetaObject FallbackGetMember(DynamicMetaObject target, DynamicMetaObject errorSuggestion)
    {
        var instance = Expression.New(DynamicWrapperConstructor, target.Expression);
        var indexExpression = Expression.MakeIndex(instance, DynamicWrapperTypeIndexer, new Expression[] {Expression.Constant(Name)});
        return DynamicMetaObject.Create(target.Value, indexExpression);
    }

    private class DynamicWrapper
    {
        private readonly object _item;
        private readonly Dictionary<string, PropertyInfo> _properties;

        public DynamicWrapper(object item)
        {
            _item = item;
            if (item != null && !(item is IDictionary<string, object>))
            {
                _properties = item.GetType().GetProperties()
                    .ToDictionary(p => p.Name, d => d);
            }
        }

        public object this[string value]
        {
            get
            {
                if (_item is null)
                    throw new NullReferenceException();
                
                if (_item is IDictionary<string, object> dictionary)
                    return dictionary[value];

                if (_properties.TryGetValue(value, out var property))
                    return property.GetValue(_item, null);

                throw new InvalidOperationException();
            }
        }
    }
}
```

**Comment by JonathanMagnan** [comment]
Hello @StefH ,

The `PR` looks good to me.

**Pull request merged**